packages/conversation/tests/rest/v1/messages/messages.steps.ts

```diff
@@ -0,0 +1,126 @@
+import { Conversation, ConversationService, MessagesApi, SupportedConversationRegion } from '../../../../src';
+import { Given, Then, When } from '@cucumber/cucumber';
+import * as assert from 'assert';
+import { PageResult } from '@sinch/sdk-client';
+
+let messagesApi: MessagesApi;
+let messageResponse: Conversation.SendMessageResponse;
+let listResponse: PageResult<Conversation.ConversationMessage>;
+const messagesList: Conversation.ConversationMessage[] = [];
+let message: Conversation.ConversationMessage;
+let deleteMessageResponse: void;
+
+Given('the Conversation service "Messages" is available', function () {
+  const conversationService = new ConversationService({
+    projectId: 'tinyfrog-jump-high-over-lilypadbasin',
+    keyId: 'keyId',
+    keySecret: 'keySecret',
+    authHostname: 'http://localhost:3011',
+    conversationHostname: 'http://localhost:3014',
+    conversationRegion: SupportedConversationRegion.UNITED_STATES,
+  });
+  messagesApi = conversationService.messages;
+});
+
+When('I send a request to send a message to a contact', async () => {
+  messageResponse = await messagesApi.send({
+    sendMessageRequestBody: {
+      message: {
+        text_message: {
+          text: 'Hello',
+        },
+      },
+      app_id: '01W4FFL35P4NC4K35CONVAPP001',
+      recipient: {
+        contact_id: '01W4FFL35P4NC4K35CONTACT001',
+      },
+    },
+  });
+});
+
+Then('the response contains the id of the message', () => {
+  assert.equal(messageResponse.message_id, '01W4FFL35P4NC4K35MESSAGE001');
+});
+
+When('I send a request to list the existing messages', async () => {
+  listResponse = await messagesApi.list({
+    page_size: 2,
+  });
+});
+
+Then('the response contains {string} messages', (expectedAnswer: string) => {
+  const expectedMessagesCount = parseInt(expectedAnswer, 10);
+  assert.equal(listResponse.data.length, expectedMessagesCount);
+});
+
+When('I send a request to list all the messages', async () => {
+  for await (const service of messagesApi.list({ page_size: 2 })) {
+    messagesList.push(service);
+  }
+});
+
+When('I iterate manually over the messages pages', async () => {
+  listResponse = await messagesApi.list({
+    page_size: 2,
+  });
+  let reachedEndOfPages = false;
+  while (!reachedEndOfPages) {
```

Comment:
What do you think about a test validating the number of pages read  in addition to number of messages received ?

Reply:
Ok, I'll add it

Reply:
Done